tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,375 @@
+        logger.info('Creating PVC with accessModes: ReadWriteOnce')
+        pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pvc'
+        )
+        pvc_data['metadata']['namespace'] = self.namespace
+        pvc_data['spec']['storageClassName'] = sc_obj.name
+        pvc_data['spec']['accessModes'] = ['ReadWriteOnce']
+        pvc_obj = PVC(**pvc_data)
+        pvc_obj.create()
+
+        # Create two pods using same PVC
+        logger.info(f"Creating two pods using same PVC {pvc_obj.name}")
+
+        # Create first pod
+        if interface_type == constants.CEPHBLOCKPOOL:
+            pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+        elif interface_type == constants.CEPHFILESYSTEM:
+            pod_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_POD_YAML
+            )
+
+        pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pod'
+        )
+        pod_data['metadata']['namespace'] = self.namespace
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = (
+            pvc_obj.name
+        )
+        pod_obj = helpers.create_pod(**pod_data)
+
+        node_pod1 = pod_obj.get()['spec']['nodeName']
+
+        # Create second pod
+        # Try creating pod until it is on a different node than first pod
+        for retry in range(1, 6):
+            pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+                'test', 'pod'
+            )
+            pod_obj2 = helpers.create_pod(wait=False, **pod_data)
+            node_pod2 = pod_obj2.get()['spec']['nodeName']
+            if node_pod1 != node_pod2:
+                break
+            logger.info(
+                f"Both pods are on same node. Deleting second pod and "
+                f"creating another pod. Retry count:{retry}"
+            )
+            pod_obj2.delete()
+            if retry == 5:
+                raise UnexpectedBehaviour(
+                    "Second pod is always created on same node as of first "
+                    "pod even after trying 5 times."
+                )
+
+        # Run IO on first pod
+        logger.info(f"Running IO on first pod {pod_obj.name}")
+        pod_obj.run_io('fs', '512M')
+        logging.info(f"Waiting for IO results from pod {pod_obj.name}")
+        fio_result = pod_obj.get_fio_results()
+        logging.info("IOPs after FIO:")
+        logging.info(
+            f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+        )
+        logging.info(
+            f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+        )
+
+        # Fetch usage details
+        mount_point = pod_obj.exec_cmd_on_pod(command="df -kh")
+        mount_point = mount_point.split()
+        usage = mount_point[mount_point.index('/var/lib/www/html') - 1]
```

Comment:
It may be good idea to check if this is None/null. We expect some usage here. If it returns nothing then there something wrong. we don't have to move forward from here if it doesn't return right value...

Reply:
Which of the params can be None?
as I see this code is very fragile so it will raise an exception anyway

Reply:
Raz, I was mentioning about this line: usage = mount_point[mount_point.index('/var/lib/www/html') - 1]

Reply:
ack

Reply:
@hnallurv, ideally we cannot get NULL as when a rbd image is mounted, it will by default use up some MBs atleast ( due to creation of an xfs, most probably ~32 MB), even though we didnt write anything. 

But yes, its good to have a check.

Reply:
Added new function to check data integrity.
Let me know if it looks good or we want earlier mount point usage check.